File: 151.reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        ans = ""
        s = s.strip()
        beg = len(s) - 1
        for forward, ch in reversed(list(enumerate(s))):
            if ch == " ":
                if beg == forward:
                    beg -= 1
                else:
                    ans += s[forward+1:beg+1] + " "
                    beg = forward - 1
        ans += s[forward:beg+1]
        return ans

File: 151.reverse-words-in-a-string.c
#define SPACE ' '

void copyToken(char* from, int beg, int end, char* to) {
	int i = strlen()

}

char * reverseWords(char * s){
	int len, forward, beg;
	len = strlen(s);

	for (forward = beg = len-1; forward >= 0; forward--) {
		if (s[forward] == SPACE) {
			if (beg == forward) {
				beg--;
			}
			else {
				copyToken(s, forward+1, beg);
				beg = forward;
			}
		}
	}
}
